crates/katana/node/src/lib.rs

```diff
@@ -206,18 +224,20 @@ pub async fn build(mut config: Config) -> Result<Node> {
     // --- build l1 gas oracle
 
     // Check if the user specify a fixed gas price in the dev config.
+    // cases to cover:
+    //      1. Fixed price by user
+    //      2. No fixed price by user and no sampling
+    //      3. Sampling with user input provider url
     let gas_oracle = if let Some(fixed_prices) = config.dev.fixed_gas_prices {
         L1GasOracle::fixed(fixed_prices.gas_price, fixed_prices.data_gas_price)
-    }
-    // TODO: for now we just use the default gas prices, but this should be a proper oracle in the
-    // future that can perform actual sampling.
-    else {
+    } else if config.gas_price_worker.as_ref().map_or(false, |worker| worker.no_sampling) {
         L1GasOracle::fixed(
             GasPrices { eth: DEFAULT_ETH_L1_GAS_PRICE, strk: DEFAULT_STRK_L1_GAS_PRICE },
             GasPrices { eth: DEFAULT_ETH_L1_DATA_GAS_PRICE, strk: DEFAULT_STRK_L1_DATA_GAS_PRICE },
         )
+    } else {
+        L1GasOracle::sampled()
```

Comment:
we can provide the l1 provider url directly in this phase

```suggestion
        L1GasOracle::sampled(l1_provider_url)
```

and lets include the url directly in the `SampledL1GasOracle` struct